### ISSUE TITLE
RC1 testing

### ISSUE THREAD

**Issue by shprink** [shprink/ionic-native-transitions#25]
Guys, I have released v1.0.0-rc1 with your requested features:

* Success/Error events
* Change page programatically (stateGo, locationUrl functions)
* Change page with directives
* Enable/Disable as a service
* Enable/Disable ionic transitions

Can you check if it is all good for you? If I do not have negative feedback I will release v1.0.0 :)

@hlehmann @leobetosouza  @mabdelfattah @NgYueHong 

PS: Do not forget to use the telerik plugin v.0.5.4

**Comment by NgYueHong** [comment]
@shprink Wow, Great work and big improvement update!! I really like the new `stateGo`, `locationUrl` and the directive feature. I am looking for it as well.

However, here I would like to ask for adding back `pendingTransitionDelay` setting which was once introduced before. Yeah, having that settings may sound stupid as we now have full control when to animate it. But I wish to have that setting as optional and able to apply on certain state / page. Because there are still cases where we might need to have some extra small delay on heavy pages to have smoother and better transition on slow devices.

**Comment by shprink** [comment]
Yeah I could add it back.

**Comment by NgYueHong** [comment]
@shprink  Can we have custom `triggerTransitionEvent` on certain state? If can be done, maybe can document it.

**Comment by shprink** [comment]
It is not possible right now but I guess it is technically possible.

**Comment by tbro028** [comment]
A nice feature might be to have the 'back' button do the opposite of the what was used to enter the view. 

If the page was slide left to enter, back could be slide right, if it was slide up, slide down, flip --> flip... etc etc

**Comment by shprink** [comment]
Yeah but you can already set default back and forward transitions @tbro028

**Comment by NgYueHong** [comment]
@shprink I guess what @tbro028 mean is to have a function that will automatically do opposite transition. For example if this is the series of animation I go through when enter to next page:
slide right, slide right, slide up, fade in

Then, if now I continuously hit the back button, it would be:
fade out, slide down, slide left, slide left

It maybe hard to have it to be automatically, but maybe enable us to manually set custom back animation for certain page?

**Comment by tbro028** [comment]
exactly, because i slide into my menu items I want to slide right back out. Thats fine, but on one of those pages i have a button which flips to a page, i'd like to be able to flip back out of it. Likewise I have a button which slides up a page, i'd like to slide that back down like a modal if they want to exit it

**Comment by shprink** [comment]
> but maybe enable us to manually set custom back animation for certain page?

I could add that that's a great feature to have.

Also I could add the automatic opposite back transition. I do not think it is complecated. Thanks for sharing those use cases guys :)

**Comment by shprink** [comment]
Apparently there is a bug on iOS swipe back #26
Need to test

**Comment by shprink** [comment]
Anyone else is having #24 issue?

**Comment by shprink** [comment]
Rc2 is out with now the ability for the hardware back button to close modal, menu or action sheet if open instead of transitioning back. #32 

It mimics the actual ionic behaviour.

**Comment by NgYueHong** [comment]
How 'bout custom `triggerTransitionEvent`?

**Comment by shprink** [comment]
Sorry, I forgot @NgYueHong

**Comment by shprink** [comment]
#36 @NgYueHong

**Comment by shprink** [comment]
R3 released.

* [FEATURE] Opposite transition on back button (backInOppositeDirection: true) <https://github.com/shprink/ionic-native-transitions/issues/27>
* [FEATURE] Back transition per state <https://github.com/shprink/ionic-native-transitions/issues/28>

**Comment by tbro028** [comment]
Quick test. Opposite back doesn't work on hardware back button. App btn ok.

**Comment by shprink** [comment]
@tbro028 are you using crosswalk? I have no problem with it.

**Comment by tbro028** [comment]
No, will try crosswalk and let you know

**Comment by tbro028** [comment]
Same problem in the 1.4.0 version of pgb-adapt crosswalk plugin. Which is
using xwalk 15 but def not the latest version as it has the keyboard
flicker.

**Comment by shprink** [comment]
v1.0.0-rc4 is out. it attempts to fix #39

Let me know @tbro028

**Comment by tbro028** [comment]
rc5 working for #39!!!!!!!!!!!!

**Comment by shprink** [comment]
Awesome, I guess only the ios swipe back is missing for the stable release. If any of you have any idea how we can simply do that please comment on #41

**Comment by shprink** [comment]
Guys can you reproduce this: #47

For me when I hit the back button on Android it closes the app if there is no more history.

**Comment by shprink** [comment]
RC6 is released!

https://github.com/shprink/ionic-native-transitions/blob/develop/CHANGELOG.md#100-rc6

**Comment by tbro028** [comment]
I had the problem, but I wanted to have a popup to confirm if user wanted
to exit the app so I override the hardware back button and implement my own
exit strategy so it's no longer an issue.

**Comment by michaeldisaro** [comment]
I just tried the plugin, when I tap on the button to change the state the transition works well.

The problem is that it's also triggering the datepicker opening on the next view because the datepicker it's on the same position of the button in the previous page.

The result is that I arrive on the page with the datepicker already opened like I had tapped on it.

Do i miss something in configuration?

**Comment by shprink** [comment]
@michaeldisaro there is a ticket for that => https://github.com/shprink/ionic-native-transitions/issues/70

**Comment by michaeldisaro** [comment]
Sorry I did not see that.
I'm not a native android developer, but looking in the code made me think about the fact that you are pre-creating a ViewGroup during init with some pre-rendered views on which you do animations. 
The default behaviour of view group, I think (excuse me if I'm wrong, you surely got more experience than me with android), is that it propagates touch events to all its childs if you do not override onInterceptTouchEvent() properly. That may be the reason why a touch also triggers event on next view.

**Comment by shprink** [comment]
@michaeldisaro maybe that something you want to ask @EddyVerbruggen the creator of the cordova plugin https://github.com/Telerik-Verified-Plugins/NativePageTransitions

**Comment by michaeldisaro** [comment]
Yes, you're right, you just made the ionic integration.
